### PR TITLE
DATA-1246 process review wherescape os

### PR DESCRIPTION
**Pull request by BFJonk**
Processed review of pull request #6 : https://github.com/wearespindle/python-wherescape/pull/6

### CODE REVIEW

File: validators/fact_dimension_join.py [hunk 1]
```diff
@@ -1,86 +1,97 @@
...
 
-    Creates a file in  output_file_location
+    Creates a file in output_file_location (WSL_WORKDIR by default)
     """
     wherescape = WhereScape()
 
     sql = """
+    -- Get table names and column names from WhereScape repository   
+    --     that contain dimension keys from all fact tables 
     select dt_schema, ft_table_name, fc_col_name
```
Comment (Shrikey) [inline validators/fact_dimension_join.py]: Pretty sql please.

File: validators/fact_dimension_join.py [hunk 2]
```diff
@@ -1,86 +1,97 @@
...
-    -- and fc_col_name like 'dim_%'
-     order by 1,2
+        left join dbo.ws_fact_tab on fc_obj_key = ft_obj_key
+        left join dbo.ws_obj_object on oo_obj_key = ft_obj_key
+        left join dbo.ws_dbc_target on dt_target_key = oo_target_key
+     where
+       -- key_type is also shown in wherescape UI. Alternative would have been
+       --   where fc_col_name like 'dim_%'
+       fc_key_type = '2'
+    order by 1,2
```
Comment (Shrikey) [inline validators/fact_dimension_join.py]: Order by column name

File: validators/fact_dimension_join.py [hunk 3]
```diff
@@ -1,86 +1,97 @@
...
-      left join dbo.ws_obj_object on oo_obj_key = ft_obj_key
-      left join dbo.ws_dbc_target on dt_target_key = oo_target_key
-     where fc_key_type = '2'
-    -- and fc_col_name like 'dim_%'
-     order by 1,2
+        left join dbo.ws_fact_tab on fc_obj_key = ft_obj_key
+        left join dbo.ws_obj_object on oo_obj_key = ft_obj_key
+        left join dbo.ws_dbc_target on dt_target_key = oo_target_key
+     where
+       -- key_type is also shown in wherescape UI. Alternative would have been
```
Comment (Shrikey) [inline validators/fact_dimension_join.py]: I would just leave this out. This is much better then the alternative.

File: wherescape/ws_env_template.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    # -- script can run in all environments
+    os.environ["WSWH_L_ETL"] = "__pasword__"
+    os.environ["WSWH_D1_ETL"] = "__pasword__"
+    os.environ["WSWH_D2_ETL"] = "__pasword__"
+    os.environ["WSWH_A_ETL"] = "__pasword__"
+    os.environ["WSWH_P_ETL"] = "__pasword__"
+
+
+if __name__ == "__main__":
+    setup_env("load_hsv_feedback_submissions")
```
Comment (Shrikey) [inline wherescape/ws_env_template.py]: This env name seems wrong

File: validators/fact_dimension_join.py (hunk 1 above)
Comment (BFJonk) [reply validators/fact_dimension_join.py]: fixed

File: validators/fact_dimension_join.py (hunk 2 above)
Comment (BFJonk) [reply validators/fact_dimension_join.py]: left the alternative out

File: wherescape/ws_env_template.py (hunk 4 above)
Comment (BFJonk) [reply wherescape/ws_env_template.py]: Fixed. Clarified calling the function with more generic parameters.

**Pull request merged**